crates/forge/tests/fixtures/zk/ScriptSetup.s.sol

```diff
@@ -0,0 +1,26 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+import {Script} from "forge-std/Script.sol";
+import {Greeter} from "../src/Greeter.sol";
+
+contract ScriptSetupNonce is Script {
+    address alice;
+
+    function setUp() public {
+        // Perform transactions and deploy contracts in setup to increment nonce and verify broadcast nonce matches onchain
+        new Greeter();
+        new Greeter();
+        alice = makeAddr("alice");
+        (bool success,) = address(alice).call{value: 1 ether}("");
+        require(success, "Failed to send ether");
+    }
+
+    function run() public {
+        vm.startBroadcast();
+        Greeter greeter = new Greeter();
+        greeter.greeting("john");
+        assert(address(alice).balance == 1 ether);
+        vm.stopBroadcast();
+    }
+}
```

Comment:
Thinking if we should add a simpler test as well. Basically a `forge test` that does the same in `setUp` but verifies if `nonce` is same in `testZkNonceIsZeroInTest()` via `vm.getNonce()`.

Reply:
Yeah I agree @Jrigada, I actually don't think we _need_ the script, in the end we just want to check that now tx nonce doesn't increase each call/create (regardless of setUp or not)

Reply:
Agree with the test, I will add a solidity test also while keeping this script just because the initial bug was discovered with a user script. Is that alright?